big_tests/tests/graphql_offline_SUITE.erl

```diff
@@ -0,0 +1,180 @@
+    escalus:fresh_story_with_config(Config, [{mike, 1}, {kate, 1}], fun admin_delete_expired_messages2_test/3).
+
+admin_delete_expired_messages2_test(Config, JidMike, JidKate) ->
+    generate_message(JidMike, JidKate, 10, 2),
+    generate_message(JidMike, JidKate, 10, 2),
+    Vars = #{<<"domain">> => domain()},
+    GraphQlRequest = admin_send_request(Config, Vars, fun admin_delete_expired_messages/0),
+    Message = ok_result(<<"offline">>, <<"deleteExpiredMessages">>, GraphQlRequest),
+    ?assertEqual(<<"Removed 2 messages">>, Message).
+
+admin_delete_old_messages2_test(Config) ->
+    escalus:fresh_story_with_config(Config, [{mike, 1}, {kate, 1}], fun admin_delete_old_messages2_test/3).
+
+admin_delete_old_messages2_test(Config, JidMike, JidKate) ->
+    generate_message(JidMike, JidKate, 2, 10),
+    generate_message(JidMike, JidKate, 2, 10),
+    Vars = #{<<"domain">> => domain(), <<"days">> => 2},
+    GraphQlRequest = admin_send_request(Config, Vars, fun admin_delete_old_messages/0),
+    Message = ok_result(<<"offline">>, <<"deleteOldMessages">>, GraphQlRequest),
+    ?assertEqual(<<"Removed 2 messages">>, Message).
+
+admin_delete_expired_messages_no_domain_test(Config) ->
+    Vars = #{<<"domain">> => <<"AAAA">>},
+    GraphQlRequest = admin_send_request(Config, Vars, fun admin_delete_expired_messages/0),
+    ParsedResult = error_result(<<"extensions">>, <<"code">>, GraphQlRequest),
+    ?assertEqual(<<"domain_not_found">>, ParsedResult).
+
+admin_delete_old_messages_no_domain_test(Config) ->
+    Vars = #{<<"domain">> => <<"AAAA">>, <<"days">> => 2},
+    GraphQlRequest = admin_send_request(Config, Vars, fun admin_delete_old_messages/0),
+    ParsedResult = error_result(<<"extensions">>, <<"code">>, GraphQlRequest),
+    ?assertEqual(<<"domain_not_found">>, ParsedResult).
+
+admin_delete_expired_messages_offline_not_configured_test(Config) ->
+    Vars = #{<<"domain">> => domain()},
+    GraphQlRequest = admin_send_request(Config, Vars, fun admin_delete_expired_messages/0),
+    ParsedResult = error_result(<<"extensions">>, <<"code">>, GraphQlRequest),
+    ?assertEqual(<<"module_not_loaded_error">>, ParsedResult).
+
+admin_delete_old_messages_offline_not_configured_test(Config) ->
+    Vars = #{<<"domain">> => domain(), <<"days">> => 2},
+    GraphQlRequest = admin_send_request(Config, Vars, fun admin_delete_old_messages/0),
+    ParsedResult = error_result(<<"extensions">>, <<"code">>, GraphQlRequest),
+    ?assertEqual(<<"module_not_loaded_error">>, ParsedResult).
+
+% Helpers
+
+admin_delete_expired_messages() ->
+    <<"mutation M1($domain: String!)
+           {offline{deleteExpiredMessages(domain: $domain)}}">>.
+
+admin_delete_old_messages() ->
+    <<"mutation M1($domain: String!, $days: Int!)
+           {offline{deleteOldMessages(domain: $domain, days: $days)}}">>.
+
+admin_send_request(Config, Vars, Function) ->
+    Body = #{query => Function(), operationName => <<"M1">>, variables => Vars},
```

Comment:
I don't see a benefit of passing a function here - a query could be passed as well, and this would be much easier to debug.